torax/_src/config/build_runtime_params.py

```diff
@@ -84,7 +89,7 @@ def from_config(
         time_step_calculator=config.time_step_calculator,
     )
 
-  @jax.jit
+  # @jax.jit
```

Comment:
I left this here as I wasn't sure whether there was a particular reason this was jitted originally, or whether it was because the provider was reused between the initial state and the simulation.
If we want to keep it jitted, there's another cache miss due to weak typing that needs hunting down!

Reply:
Great, then I think we can leave a TODO to look into adding JIT there plus resolving the cache miss but makes sense to remove it at this time.
(It might make a difference to compile time even if it's only under other JITs because subcompilation boundaries are applied wherever we use the `jax.jit` with the XLA flatten flag we have applied but that can be looked at later.)